protocols/relay/src/lib.rs

```diff
@@ -23,9 +23,48 @@
 
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 
+mod behaviour;
+mod copy_future;
+mod priv_client;
+mod protocol;
 pub mod v2;
 
+#[allow(clippy::derive_partial_eq_without_eq)]
+mod message_proto {
+    include!(concat!(env!("OUT_DIR"), "/message_v2.pb.rs"));
+}
+
+pub use behaviour::{Behaviour, CircuitId, Config, Event};
+pub use protocol::{
+    inbound_hop::FatalUpgradeError as InboundHopFatalUpgradeError,
+    inbound_stop::FatalUpgradeError as InboundStopFatalUpgradeError,
+    outbound_hop::FatalUpgradeError as OutboundHopFatalUpgradeError,
+    outbound_stop::FatalUpgradeError as OutboundStopFatalUpgradeError, HOP_PROTOCOL_NAME,
+    STOP_PROTOCOL_NAME,
+};
+
+// explicitly pub'ed client types.
+pub mod client {
+    pub use crate::priv_client::{Behaviour, Event, RelayedConnection, Transport};
+
+    pub mod transport {
+        pub use crate::priv_client::transport::{
+            Error, Listener, Reservation, ToListenerMsg, TransportToBehaviourMsg,
```

Comment:
Do all of these have to be public for the user to interact with `libp2p-relay`?

Reply:
Good question Thomas, I am not yet familiar with the `relay` and `dcutr` protocols. But taking into consideration that `dcutr` uses `relay` and doesn't use any of the `transport::` types, I'd wonder if they are required to be public at all? 
What if we deprecate them and see if anyone complains? CC @mxinden

Reply:
- When these protocols were merged, `Listener` was still in the public API of `Transport` but it no longer is.
- I wonder if the messages show up in some other public API.
- I think it makes sense for the error to be exposed.

Deprecating them with a notice that they will be private in the future sounds good to me!

Reply:
ok nice, updated! Ptal Thomas if you agree with the wording.

Reply:
Thanks @jxs and @thomaseizinger. Deprecation sounds good to me.